engine/src/main/java/org/terasology/logic/players/LocalPlayerSystem.java

```diff
@@ -212,6 +215,33 @@ private void standPlayer(EntityRef entity) {
         clientComp.character.send(new SetMovementModeEvent(MovementMode.WALKING));
     }
 
+    /**
+     * Auto move is disabled every time any other movement event occurs.
+     */
+    private void stopAutoMove() {
+        isAutoMove = false;
+    }
+
+    /**
+     * Start a thread that moves the entity in the camera direction at every period of 0.2 seconds.
+     * @param entity
+     */
+    private void startAutoMove(EntityRef entity) {
+        isAutoMove = true;
+        ClientComponent clientComponent = entity.getComponent(ClientComponent.class);
+        new Thread(() -> {
```

Comment:
@zeokav I have some usage of DelayManager in [one of my repos](https://github.com/oniatus/UniqueServerNames/blob/master/src/main/java/org/terasology/uniqueservernames/system/ServerUniqueNameSystem.java).
Check out Line 40 (get an instance), Line 78 (register delayed action) and Line 82F (handle the event). I think there are two different events for delayed and periodic action, so if you need the other one, you may want to inspect the source of delaymanager directly :)